### PR TITLE
Dl user story 21

### PR DESCRIPTION
**Pull request by dlayton66**
This PR:

- adds the table_print gem to make rails console output easier to read (http://tableprintgem.com/)
- adds FactoryBot to create testing data
- completes User Story 21

### CODE REVIEW

File: spec/models/customer_spec.rb [hunk 1]
```diff
@@ -1,7 +1,40 @@
...
+    FactoryBot.create_list(:transaction, 3, result: 'success', invoice_id: @invoices[4].id)
+    FactoryBot.create_list(:transaction, 2, result: 'success', invoice_id: @invoices[2].id)
+  end
+
   describe "Relationships" do
     it {should have_many :invoices} 
+    it {should have_many(:transactions).through(:invoices)}
+  end
+
+  describe '#successful_transactions' do
```
Comment (kaylahrose) [inline spec/models/customer_spec.rb:20]: Would this follow SRP more closely if it was a transaction model method?

Comment (kaylahrose) [reply spec/models/customer_spec.rb:20]: Actually, this fulfills the user story well here. 👏🏽

**Review by kaylahrose — APPROVED** [review]
LGTM

File: spec/models/customer_spec.rb (hunk 1 above)
Comment (dlayton66) [reply spec/models/customer_spec.rb:20]: It definitely might!  We'll see what Brian says - I'll circle back to refactor if need be

**Pull request merged**